File: Assets/Scripts/Game/Animator/AnimationLibrary.cs
public static class AnimationLibrary
{
    private static Dictionary<string, Func<GameObject, AnimatorGraph>> animations = new()
    {
        {"Example", CreateExampleAnimations}
    };

    public static AnimatorGraph GetAnimator(string type, GameObject obj)
    {
        if (animations.TryGetValue(type, out var factory))
        {
            return factory(obj);
        }
        throw new ArgumentException($"Unknown animation type: {type}");
    }

    private static AnimatorGraph CreateExampleAnimations(GameObject obj)
    {
        AnimatorGraph animator = new AnimatorGraph("Idle");

        var testUpd = obj.GetComponent<TestObjUpdater>() as TestObjUpdater;
        if (testUpd == null)
        {
            testUpd = new TestObjUpdater();
            obj.addUpdater(testUpd);
        }

        animator.AddTransition("Idle", "Walk", () => testUpd.isWalking);
        animator.AddTransition("Walk", "Idle", () => !testUpd.isWalking);

        return animator;
    }
}

File: wwwroot/js/AnimationLibrary.js
import { Animation } from './Animation.js';
import { NumericTrack } from './NumericTrack.js';
import { SpriteTrack } from './SpriteTrack.js';

export class AnimationLibrary {
    static animations = {
        "Example": AnimationLibrary.createExampleAnimations()
    };

    static getAnimations(type) {
        return this.animations[type];
    }

    static createExampleAnimations() {
        return {
            "Idle": new Animation([
                new NumericTrack("scale.y", [
                    { time: 0, value: 1 },
                    { time: 0.2, value: 1.2 },
                    { time: 0.4, value: 1 }
                ])
            ], true),
            "Walk": new Animation([
                new NumericTrack("rotation", [
                    { time: 0, value: 0 },
                    { time: 2, value: Math.PI * 2 }
                ]) 
            ], true)
        };
    }

}